src/Symfony/Component/Security/Csrf/SameOriginCsrfTokenManager.php

```diff
@@ -235,6 +235,10 @@ public function onKernelResponse(ResponseEvent $event): void
      */
     private function isValidOrigin(Request $request): ?bool
     {
+        if ($request->headers->has('Sec-Fetch-Site')) {
+            return 'same-origin' === $request->headers->get('Sec-Fetch-Site');
+        }
```

Comment:
Nit: prevent an extra call to `has()` on the hot path, as most requests will have this header set.

```suggestion
        $secFetchSite = $request->headers->get('Sec-Fetch-Site');
        if ($secFetchSite !== null)) {
            return 'same-origin' === $secFetchSite;
        }
```

Reply:
updated thanks